lib/image-stream/README.md

```diff
@@ -0,0 +1,57 @@
+Etcher Image Stream
+===================
+
+This module is in charge of creating a readable stream from any image source
+(e.g: a file, a URL, etc) along with some metadata (like size), and handling
+any necessary transformations (like decompression) that must be applied before
+plugging the stream to [`etcher-image-write`][etcher-image-write].
+
+Given that this module contains the logic to handle image formats, the module
+becomes the most reliable source of truth for the list of supported ones.
+
+There are three classes of images this module supports:
+
+- Uncompressed images (e.g: `.img`, `.iso`)
+- Compressed images (e.g: `.img.xz`, `.iso.gz`)
+- Archive images (e.g: `.zip`)
+
+The core of this module consists of handlers and archive hooks.
+
+Handlers
+--------
+
+The handlers are functions that know how to handle certain MIME types, like
+`application/x-bzip2` and `application/octet-stream`, returning a stream for
+the image, a transform stream that needs to be applied to get the real image
+data, and useful metadata like the final image size.
+
+Archive Hooks
+-------------
+
+This module supports reading "archive images", which are defined by handlers
+(like `application/zip`). In order to avoid duplication on how to handle
+archives, archive support is implemented by "archive hooks".
+
+Archive hooks are CommonJS modules that expose two functions:
```

Comment:
Could you also add how, and on what the archive hook modules would be instantiated here (i.e. are they called on the stream, or file descriptor, ...)? Or is that too much information, which should rather be somewhere else?